### PR TITLE
feat: add org_uuid filter

### LINKED ISSUE #1311: Enable search endpoint in org homepage
### Context
We will enable searching of bounties in org homepage. Let's enable the endpoint for this. 

Use same endpoint we use on the homepage search bar and add query params for org_uuid. 
Organization Page-Desktop (1)

### Acceptance Criteria
- [ ] Create a unit test so that if you do search with the new query parameters, that only bounties associated with that org show up.

### PR DESCRIPTION
**Pull request by wright-eric**
## Describe your changes
added org_uuid filter on bounties endpoint

## Issue ticket number and link
close #1311 

## Type of change
new feature

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox

### CODE REVIEW

**Comment by ecurrencyhodler** [comment]
All checks passed. Let's get a code review.

**Comment by kevkevinpal** [comment]
Are you able to add a test to check if this filter works? @wright-eric

**Comment by wright-eric** [comment]
> Are you able to add a test to check if this filter works? @wright-eric

I was not able to add unit test because the gorm.DB id a struct and not interface. I tested by connecting to my local database. 

Maybe for DB function we can have integration tests. Connecting to actual DB and then checking. Can setup a job in github actions with postgres setup and all. And then can test that

**Comment by wright-eric** [comment]
## Query with org filter
curl 'http://localhost:5002/gobounties/all?limit=10&sortBy=created&search=&page=1&resetPage=true&Open=true&Assigned=false&Paid=false&org_uuid=cmd9nvaku6rr86n6ank0'    --compressed | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1542  100  1542    0     0   190k      0 --:--:-- --:--:-- --:--:--  215k
[
  {
    "bounty": {
      "id": 2,
      "owner_id": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "paid": false,
      "show": true,
      "type": "freelance_job_request",
      "award": "",
      "assigned_hours": 0,
      "bounty_expires": "",
      "commitment_fee": 0,
      "price": 1000000,
      "title": "Test 2",
      "tribe": "None",
      "assignee": "",
      "ticket_url": "",
      "org_uuid": "cmd9nvaku6rr86n6ank0",
      "description": "test 2",
      "wanted_type": "Design",
      "deliverables": "",
      "github_description": false,
      "one_sentence_summary": "",
      "estimated_session_length": "",
      "estimated_completion_date": "",
      "created": 1704634165,
      "updated": "2024-01-07T19:09:54.530812+05:30",
      "coding_languages": []
    },
    "assignee": {
      "id": 0,
      "uuid": "",
      "owner_pubkey": "",
      "owner_alias": "",
      "unique_name": "",
      "description": "",
      "tags": null,
      "img": "",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "owner": {
      "id": 2,
      "uuid": "cmd9ftqku6rrt45fip7g",
      "owner_pubkey": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "owner_alias": "Jesse",
      "unique_name": "jesse",
      "description": "Im cool",
      "tags": [
        "tag1"
      ],
      "img": "https://cliparting.com/wp-content/uploads/2018/03/cool-pictures-2018-2.jpg",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "organization": {
      "uuid": "cmd9nvaku6rr86n6ank0",
      "name": "Guardians",
      "img": ""
    }
  }
]



## without org filter
curl 'http://localhost:5002/gobounties/all?limit=10&sortBy=created&search=&page=1&resetPage=true&Open=true&Assigned=false&Paid=false'   --compressed | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4559    0  4559    0     0   520k      0 --:--:-- --:--:-- --:--:--  556k
[
  {
    "bounty": {
      "id": 4,
      "owner_id": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "paid": false,
      "show": true,
      "type": "freelance_job_request",
      "award": "",
      "assigned_hours": 0,
      "bounty_expires": "",
      "commitment_fee": 0,
      "price": 1212121212,
      "title": "New without org 2",
      "tribe": "None",
      "assignee": "",
      "ticket_url": "",
      "org_uuid": "",
      "description": "dawdawdawd",
      "wanted_type": "Dev ops",
      "deliverables": "",
      "github_description": false,
      "one_sentence_summary": "",
      "estimated_session_length": "",
      "estimated_completion_date": "",
      "created": 1704635909,
      "updated": "2024-01-07T19:28:29.325026+05:30",
      "coding_languages": []
    },
    "assignee": {
      "id": 0,
      "uuid": "",
      "owner_pubkey": "",
      "owner_alias": "",
      "unique_name": "",
      "description": "",
      "tags": null,
      "img": "",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "owner": {
      "id": 2,
      "uuid": "cmd9ftqku6rrt45fip7g",
      "owner_pubkey": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "owner_alias": "Jesse",
      "unique_name": "jesse",
      "description": "Im cool",
      "tags": [
        "tag1"
      ],
      "img": "https://cliparting.com/wp-content/uploads/2018/03/cool-pictures-2018-2.jpg",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "organization": {
      "uuid": "",
      "name": "",
      "img": ""
    }
  },
  {
    "bounty": {
      "id": 3,
      "owner_id": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "paid": false,
      "show": true,
      "type": "freelance_job_request",
      "award": "",
      "assigned_hours": 0,
      "bounty_expires": "",
      "commitment_fee": 0,
      "price": 100000,
      "title": "bounty without org",
      "tribe": "None",
      "assignee": "",
      "ticket_url": "",
      "org_uuid": "",
      "description": "without org",
      "wanted_type": "Design",
      "deliverables": "",
      "github_description": false,
      "one_sentence_summary": "",
      "estimated_session_length": "",
      "estimated_completion_date": "",
      "created": 1704635484,
      "updated": "2024-01-07T19:21:24.574766+05:30",
      "coding_languages": []
    },
    "assignee": {
      "id": 0,
      "uuid": "",
      "owner_pubkey": "",
      "owner_alias": "",
      "unique_name": "",
      "description": "",
      "tags": null,
      "img": "",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "owner": {
      "id": 2,
      "uuid": "cmd9ftqku6rrt45fip7g",
      "owner_pubkey": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "owner_alias": "Jesse",
      "unique_name": "jesse",
      "description": "Im cool",
      "tags": [
        "tag1"
      ],
      "img": "https://cliparting.com/wp-content/uploads/2018/03/cool-pictures-2018-2.jpg",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "organization": {
      "uuid": "",
      "name": "",
      "img": ""
    }
  },
  {
    "bounty": {
      "id": 2,
      "owner_id": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "paid": false,
      "show": true,
      "type": "freelance_job_request",
      "award": "",
      "assigned_hours": 0,
      "bounty_expires": "",
      "commitment_fee": 0,
      "price": 1000000,
      "title": "Test 2",
      "tribe": "None",
      "assignee": "",
      "ticket_url": "",
      "org_uuid": "cmd9nvaku6rr86n6ank0",
      "description": "test 2",
      "wanted_type": "Design",
      "deliverables": "",
      "github_description": false,
      "one_sentence_summary": "",
      "estimated_session_length": "",
      "estimated_completion_date": "",
      "created": 1704634165,
      "updated": "2024-01-07T19:09:54.530812+05:30",
      "coding_languages": []
    },
    "assignee": {
      "id": 0,
      "uuid": "",
      "owner_pubkey": "",
      "owner_alias": "",
      "unique_name": "",
      "description": "",
      "tags": null,
      "img": "",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "owner": {
      "id": 2,
      "uuid": "cmd9ftqku6rrt45fip7g",
      "owner_pubkey": "038c3c1f4d304c7b997fecfdaf8fdfc2215405942c025349b45de9dfe6fdb8a43e",
      "owner_alias": "Jesse",
      "unique_name": "jesse",
      "description": "Im cool",
      "tags": [
        "tag1"
      ],
      "img": "https://cliparting.com/wp-content/uploads/2018/03/cool-pictures-2018-2.jpg",
      "created": null,
      "updated": null,
      "unlisted": false,
      "deleted": false,
      "last_login": 0,
      "owner_route_hint": "",
      "owner_contact_key": "",
      "price_to_meet": 0,
      "new_ticket_time": 0,
      "twitter_confirmed": false,
      "extras": null,
      "github_issues": null
    },
    "organization": {
      "uuid": "cmd9nvaku6rr86n6ank0",
      "name": "Guardians",
      "img": ""
    }
  }
]

File: db/db.go [hunk 1]
```diff
@@ -685,10 +687,13 @@ func (db database) GetAllBounties(r *http.Request) []Bounty {
...
 	}
+	if orgUuid != "" {
+		orgQuery = "AND org_uuid = " + orgUuid
+	}
 	query := "SELECT * FROM public.bounty WHERE show != false"
 
-	allQuery := query + " " + openQuery + " " + assignedQuery + " " + paidQuery + " " + searchQuery + " " + orderQuery + " " + limitQuery
-
+	allQuery := query + " " + openQuery + " " + assignedQuery + " " + paidQuery + " " + searchQuery + " " + orderQuery + " " + limitQuery + " " + orgQuery
+	fmt.Println(allQuery)
```
Comment (elraphty) [inline db/db.go]: @wright-eric remove the print line statement

Comment (wright-eric) [reply db/db.go]: I think I have already removed that print line after the latest commit

**Comment by AbdulWahab3181** [comment]
Hi @wright-eric, I noticed that you worked on the task. Could you please verify if the endpoint https://people.sphinx.chat/gobounties/all?limit=10&sortBy=created&search=&page=1&resetPage=true&Open=true&Assigned=false&Paid=false&org_uuid=cmd9nvaku6rr86n6ank0 is providing the expected result? Despite setting cmd9nvaku6rr86n6ank0 as the org_uuid, we are still receiving  other organizations in responses.

**Comment by wright-eric** [comment]
@AbdulWahab3181  i dont think it's deployed yet... You can use the staging url. 
I think it's people-test.sphinx.chat

**Comment by AbdulWahab3181** [comment]
@wright-eric Maybe, it's working with people-test.sphinx.chat. Thanks.

**Pull request merged**